### PR TITLE
Add PMIC, ADC, and GPIO Expander targets

### PR DESCRIPTION
**Pull request by RameshIyyar**

### CODE REVIEW

**Comment by amitay** [comment]
You cannot add generic i2c devices anywhere in the system tree.  If you want to use access them via i2c bus, then there needs to be a right i2c bus with driver before you can start adding i2c devices.

File: libpdbg/p10_fapi_targets.c [hunk 1]
```diff
@@ -563,6 +563,16 @@ struct nx p10_nx = {
 };
 DECLARE_HW_UNIT(p10_nx);
 
+struct adc p10_adc = {
+    .target = {
+        .name = "POWER10 Analog to Digital Converter",
```
Comment (ojayanth) [inline libpdbg/p10_fapi_targets.c:568]: Not sure this is right definition, ADC is not specific to p10 , system supports different types of ADCs' example ADC associated to APSS, and DIMM.

File: libpdbg/hwunit.h [hunk 2]
```diff
@@ -373,4 +373,9 @@ struct adc {
 };
 #define target_to_adc(x) container_of(x, struct adc, target)
 
+struct gpio_expander {
```
Comment (ojayanth) [inline libpdbg/hwunit.h:376]: Commit message : please provide the use case for this hardware.  This is also not processor specific right? genral purpose  bmc IO pin you are referring here ?

**Comment by amitay** [comment]
Are all the targets at the same level as DIMM target in P10 device tree?

All the patches require fixing spaces to tabs.

**Comment by RameshIyyar** [comment]
> Are all the targets at the same level as DIMM target in P10 device tree?

No, All these targets are placed under OCMB like MEM_PORT.

> All the patches require fixing spaces to tabs.
Sure, will fix it

File: libpdbg/p10_fapi_targets.c (hunk 1 above)
Comment (RameshIyyar) [reply libpdbg/p10_fapi_targets.c:568]: I have added the abbreviation for the ADC as like previously added.

Please give some better names if you have a different view.

**Comment by RameshIyyar** [comment]
@amitay @ojayanth addressed the comments, please take a look

**Comment by amitay** [comment]
Merged.

**Pull request closed without merging**